### PR TITLE
Simplify `contrib/backport`

### PR DESCRIPTION
**Pull request by silverwind**
This script was trying to be too smart, make it more straightforward and less error-prone so that i could be used by the backport bot too ideally.

- Always delete the backport branch so that script is idempotent in this regard
- Remove the push functionality, it's best done by the user because only they know the remote name
- Remove reading docs/config.yaml, it no longer exists
- Remove version detection, version is now a required argument
- Remove opening the pull request with xdg-open, xdg-open is not portable
- Remove continue from failed cherry pick. It's best to reset manually in this case
- Clean up the console logging

Example run:

```
$ go run ./contrib/backport --version v1.21 27503
* Backporting 27503 to origin/release/v1.21 as backport-27503-v1.21
* `git fetch origin main`
* `git fetch origin release/v1.21`
* `git branch -D backport-27503-v1.21`
* `git checkout -b backport-27503-v1.21 origin/release/v1.21`
* Attempting git cherry-pick 08efeb5cdc22d21b5ef12cc540727594a22062d1
* Amending commit to prepend `Backport #27503` to body
Backport done! You can now push it with `git push yourremote backport-27503-v1.21`
```

### CODE REVIEW

File: contrib/backport/backport.go [hunk 1]
```diff
@@ -202,41 +147,8 @@ func runBackport(c *cli.Context) error {
...
-func xdgOpen(ctx context.Context, url string) error {
-	fmt.Printf("* `xdg-open %s`\n", url)
-	out, err := exec.CommandContext(ctx, "xdg-open", url).Output()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "%s", string(out))
-		return fmt.Errorf("unable to xdg-open to %s: %w", url, err)
-	}
-	return nil
-}
+	fmt.Printf("Backport done! You can now push it with `git push yourremote %s`\n", backportBranch)
```
Comment (delvh) [inline contrib/backport/backport.go]:
```suggestion
	fmt.Printf("Backport done! You can now push it with `git push <your remote> %s`\n", backportBranch)
```

**Review by delvh — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]
Good to keep things simple

**Comment by zeripath** [comment]
This will break the `--continue` functionality.

**Comment by delvh** [comment]
Yes, that parameter was removed.
What exactly did that do?

**Comment by silverwind** [comment]
It did look for a aborted cherry-pick, but I think given that the script now deletes the backport branch every time, it likely will never encounter the situation again.

**Comment by silverwind** [comment]
The goal is to get this script into a state where the backporter bot can use it, so we can replace [most of this code](https://github.com/GiteaBot/gitea-backporter/blob/main/src/backport.ts).

**Comment by zeripath** [comment]
It already was in a state that the backporter tool could use.

By removing the `--continue` option you've made it impossible to use the script to help create backports when there is a conflict.

`--continue` simply allowed you to fix the conflicts and then restart the process.

Instead of simply removing things you should have just read or improved the documentation explaining what it was doing.

**Comment by silverwind** [comment]
I don't mind if you want to restore, but I thought this script should be idempotent and a `--continue` did not fit into the concept. The bot would only do clean checkouts and if there is a merge conflict, it has to be resolved manually in any case.

**Comment by zeripath** [comment]
Your simplification has now broken the specific help information that gitea-backport tells maintainers.

e.g. https://github.com/go-gitea/gitea/pull/27511#issuecomment-1751847931

It said how to use the script, if it wasn't clear you should have improved the message instead of simply destroying things.

**Comment by silverwind** [comment]
Right, so let's restore it.

**Pull request merged**